src/Extractor/HttpClient.php

```diff
@@ -42,11 +42,7 @@ class HttpClient
      */
     private $responseHistory;
 
-    /**
-     * @param Client $client Http client
-     * @param array  $config
-     */
-    public function __construct(Client $client, $config = [], LoggerInterface $logger = null)
+    public function __construct(ClientInterface $client, $config = [], LoggerInterface $logger = null)
```

Comment:
`Client` → `ClientInterface`  should be fine. `Client` has been a subtype since 2.0 (https://github.com/php-http/httplug/commit/6885b6c94751e64efc1473bedff1706cf86596d9).